enigma/src/main/java/cuchaz/enigma/analysis/index/JarIndex.java

```diff
@@ -40,6 +37,7 @@
 	private final Collection<JarIndexer> indexers;
 
 	private final Multimap<String, MethodDefEntry> methodImplementations = HashMultimap.create();
+	private final Map<ClassEntry, List<ParentedEntry>> childrenByClass;
```

Comment:
Since Multimap is a thing, it might actually be better to use that here since it's effectively the same but less annoying to use.

Reply:
Multimap has an important drawback, entries are not sorted anymore by indexing order (inner classes, then fields, then methods)

Reply:
There's a `ListMultimap` that uses `List` as the value type, which should work here.

Reply:
Changed to ListMultimap, thanks Juuz